vllm/entrypoints/openai/tool_parsers/mistral_tool_parser.py

```diff
@@ -70,6 +76,12 @@ def __init__(self, tokenizer: AnyTokenizer):
         self.bot_token = "[TOOL_CALLS]"
         self.bot_token_id = self.vocab.get(self.bot_token)
         self.tool_call_regex = re.compile(r"\[{.*}\]", re.DOTALL)
+        if _is_fn_name_regex_support(self.model_tokenizer):
+            self.fn_name_regex = re.compile(r'([a-zA-Z0-9_-]+)(\{.*?\})',
```

Comment:
just noticed that this regex sadly doesn't match the outer most {...} but instead only the inner-most. This means that for inputs such as:

'{"sub_dict": {....}}' it incorrectly parse it to '{"sub_dict": {....}' => and hence miss the final `{`. The corrected regex should be:

```
fn_name_regex = re.compile(r'([a-zA-Z0-9_-]+)(\{[\s\S]*?\})(?=\s*$|,|\s)', re.DOTALL)
```

Reply:
Thanks for the find. I can post a fix for this